### PR TITLE
[REVIEW] Upgrade to `arrow-8`

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR upgrades `arrow` to `8`.

### CODE REVIEW

**Comment by jakirkham** [comment]
rerun tests

**Comment by jakirkham** [comment]
rerun tests

**Comment by jakirkham** [comment]
rerun tests

**Comment by jakirkham** [comment]
Looks like packages are now installed 🎉

However there are segfaults in CI. Particularly after `ToArrowTest.NestedList` (same on all CI jobs that fail). May need a deeper dive into why that is happening

**Comment by galipremsagar** [comment]
> Looks like packages are now installed 🎉
> 
> However there are segfaults in CI. Particularly after `ToArrowTest.NestedList` (same on all CI jobs that fail). May need a deeper dive into why that is happening

Yup, that's expected with some stale arrow version lying around. This commit is needed to get past that error: https://github.com/rapidsai/cudf/pull/10816/commits/8a474c84df99cec4f20c340d74932260f828b312

**Comment by jakirkham** [comment]
Thanks Prem! 🙏

So was it just an old version or was there more going on with the segfault?

**Comment by galipremsagar** [comment]
> Thanks Prem! 🙏
> 
> So was it just an old version or was there more going on with the segfault?

Yea for some reason, if an env has an older version of arrow and conda upgrades it, we see that kind of segfaults. But when we force remove old versions before installing a new version or create a new env with new version of arrow we never run into such kind of segfaults.

**Review by ajschmidt8 — APPROVED** [review]

File: ci/cpu/build.sh [hunk 1]
```diff
@@ -51,6 +51,9 @@ if [ "$SOURCE_BRANCH" = "main" ]; then
   conda config --system --remove channels dask/label/dev
 fi
 
+gpuci_conda_retry remove --force "arrow-cpp" "pyarrow" "orc" 'arrow-cpp-proc=*=cuda'
+mamba install -y "arrow-cpp=8.0.0" "pyarrow=8.0.0" "orc" 'arrow-cpp-proc=*=cuda'
+
```
Comment (galipremsagar) [inline ci/cpu/build.sh:56]:
```suggestion
```

File: ci/gpu/build.sh [hunk 2]
```diff
@@ -75,6 +75,9 @@ gpuci_logger "Activate conda env"
 . /opt/conda/etc/profile.d/conda.sh
 conda activate rapids
 
+gpuci_conda_retry remove --force "arrow-cpp" "pyarrow" "orc" 'arrow-cpp-proc=*=cuda'
+mamba install -y "arrow-cpp=8.0.0" "pyarrow=8.0.0" "orc" 'arrow-cpp-proc=*=cuda'
+
```
Comment (galipremsagar) [inline ci/gpu/build.sh:80]:
```suggestion
```

**Comment by jakirkham** [comment]
Do we know what is going on with the CI failure here?

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**